sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java

```diff
@@ -2533,6 +2596,58 @@
             });
     }
 
+    /**
+     * Check that the additional headers field is present and add the additional auth header
+     *
+     * @param headerName name of the header to be added
+     * @param context current request context
+     *
+     * @return boolean representing the outcome of adding header operation
+     */
+    private void addSupplementaryAuthHeader(String headerName, String entity, Context context) {
+        context.getData(AZURE_REQUEST_HTTP_HEADERS_KEY)
+            .map(headers -> {
```

Comment:
I don't believe this is the right operator. Mapping is a transform, I wouldn't want it to return null if the HTTP headers existed but were another object type

```java
context.getData("foo").ifPresent(value -> {
            if (!(value instanceof HttpHeaders)) {
                return;
            }

            HttpHeaders customHttpHeaders = (HttpHeaders) headers;
            customHttpHeaders.add(headerName, entity);
        });
```

Reply:
makes sense, reduced it to ifpresent